mythril/laser/ethereum/instructions.py

```diff
@@ -353,37 +353,42 @@ def calldataload_(self, global_state):
         environment = global_state.environment
         op0 = state.stack.pop()
 
-        try:
-            offset = util.get_concrete_int(simplify(op0))
-            b = environment.calldata[offset]
-        except AttributeError:
-            logging.debug("CALLDATALOAD: Unsupported symbolic index")
-            state.stack.append(global_state.new_bitvec(
-                "calldata_" + str(environment.active_account.contract_name) + "[" + str(simplify(op0)) + "]", 256))
-            return [global_state]
-        except IndexError:
-            logging.debug("Calldata not set, using symbolic variable instead")
-            state.stack.append(global_state.new_bitvec(
-                "calldata_" + str(environment.active_account.contract_name) + "[" + str(simplify(op0)) + "]", 256))
-            return [global_state]
+        if environment.calldata_type == CalldataType.CONCRETE:
+            try:
+                offset = util.get_concrete_int(simplify(op0))
+                b = environment.calldata[offset]
+            except AttributeError:
+                logging.debug("CALLDATALOAD: Unsupported symbolic index")
+                state.stack.append(global_state.new_bitvec(
+                    "calldata_" + str(environment.active_account.contract_name) + "[" + str(simplify(op0)) + "]", 256))
+                return [global_state]
+            except IndexError:
+                logging.debug("Calldata not set, using symbolic variable instead")
+                state.stack.append(global_state.new_bitvec(
```

Comment:
This behavior was also wrong before. If an index is accessed outside of the calldata "bounds" then 0 should be returned

Can you change that as well?